docs/Advanced-Usage/Debugging-and-Profiling-on-FPGA/AutoCounter.rst

```diff
@@ -0,0 +1,135 @@
+possibly other code repositories), AutoCounter provides a filtering mechanism
+based on module granularity. As such, only cover functions that appear within
+selected modules will generate counters.
+
+The filtered modules can be indicated using one of two methods:
+
+1. A module selection annotation within the top-level configuration
+   implementation.  To use this method, add the
+   ``AutoCounterCoverModuleAnnotation`` annotation with the name of the module
+   for which you want the cover functions to be turned into AutoCounters.  The
+   following example will generate counters from cover functions within the
+   ``StreamWriter`` module:
+
+.. code-block:: scala
+
+   class FireSimDUT(implicit p: Parameters) extends Subsystem
+    with HasHierarchicalBusTopology
+    with CanHaveMasterAXI4MemPort
+    with HasPeripheryBootROM
+    with HasPeripherySerial
+    with HasPeripheryUART
+    with HasPeripheryIceNIC
+    with HasPeripheryBlockDevice
+    with HasTraceIO
+  {
+    override lazy val module = new FireSimModuleImp(this)
+      
+    chisel3.experimental.annotate(AutoCounterCoverModuleAnnotation("StreamWriter"))
+  }
+
+2. An input file with a list of module names. This input file is named
+   ``autocounter-covermodules.txt``, and includes a list of module names
+   separated by new lines (no commas).
+
+.. _autocounter-runtime-parameters:
+
+AutoCounter Runtime Parameters
+---------------------------------
+AutoCounter currently takes a single runtime configurable parameter, defined
+under the ``[autocounter]`` section in the ``config_runtime.ini`` file.  The
+``readrate`` parameter defines the rate at which the counters should be read,
+and is measured in target-cycles. Hence, if the read-rate is defined to be 100,
+the simulator will read and print the values of the counters every 100 cycles.
+By default, the read-rate is set to 0 cycles, which is equivalent to disabling
+AutoCounter.
+
+.. code-block:: ini
+
+   [autocounter]
+   # read counters every 100 cycles
+   readrate=100
+
+
+Upon setting this value, when you run a workload, an AutoCounter output file
+will be placed in the ``sim_slot_<slot #>`` directory on the F1 instance under
+the name ``AUTOCOUNTERFILE0``.
+
+.. Note:: AutoCounter is designed as a coarse-grained observability mechanism.
+      It assumes the counters will be read at intervals greater than O(10000) cycles.
+      If you intend on reading counters at a finer granularity, please consider using
+      synthesizable printfs (otherwise, simulation performance may degrade more than
+      necessary)
+
+Using TracerV Trigger with AutoCounter
+-----------------------------------------
+In order to collect AutoCounter results from only from a particular region of
+interest in the simulation, AutoCounter has been integrated with TracerV
+triggers. See the :ref:`tracerv-trigger` section for more information.
+
+
+AutoCounter using Synthesizable Printfs
```

Comment:
I still think this section should be removed. This is how legacy becomes not-legacy.

Reply:
ok i'm just going to remove it. cc @alonamid

Reply:
err actually i'm going to make an issue for this